core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -367,7 +370,11 @@ public RetryOnPrimaryException(StreamInput in) throws IOException {
 
     public interface PrimaryResult<R extends ReplicationRequest<R>> {
 
-        R replicaRequest();
+        /**
+         * Returns the replication request when the primary operation
+         * is successful otherwise returns <code>null</code>
```

Comment:
Also, with this offical way of saying "no need to replicate", I think we can remove the `executeOnReplicas` paramter to the `ReplicationOperation`. This can be done as a follow up. Can you add this to your list? :)